ksqldb-common/src/main/java/io/confluent/ksql/metrics/MetricCollectors.java

```diff
@@ -96,15 +117,51 @@ static String addCollector(final String id, final MetricCollector collector) {
     return finalId;
   }
 
-  public static void addConfigurableReporter(final KsqlConfig ksqlConfig) {
+  public static void addConfigurableReporter(
+      final KsqlConfig ksqlConfig,
+      final String kafkaClusterId
+  ) {
+    final String ksqlServiceId = ksqlConfig.getString(KsqlConfig.KSQL_SERVICE_ID_CONFIG);
     final List<MetricsReporter> reporters = ksqlConfig.getConfiguredInstances(
         KsqlConfig.METRIC_REPORTER_CLASSES_CONFIG,
-        MetricsReporter.class);
-    for (final MetricsReporter reporter: reporters) {
-      metrics.addReporter(reporter);
+        MetricsReporter.class,
+        Collections.singletonMap(
+            KsqlConfig.KSQL_SERVICE_ID_CONFIG,
+            ksqlServiceId));
+
+    if (reporters.size() > 0) {
+      final Map<String, Object> metadata =
+          new HashMap<>(ksqlConfig.originalsWithPrefix(CommonClientConfigs.METRICS_CONTEXT_PREFIX));
+      final MetricsContext metricsContext =
+          new KafkaMetricsContext(
+              KSQL_JMX_PREFIX,
+              addConfluentMetricsContextConfigs(
+                  metadata,
+                  ksqlServiceId,
+                  kafkaClusterId));
+
+      for (final MetricsReporter reporter : reporters) {
+        reporter.contextChange(metricsContext);
+        metrics.addReporter(reporter);
+      }
     }
   }
 
+  public static Map<String, Object> addConfluentMetricsContextConfigs(
+      final Map<String,Object> props,
+      final String ksqlServiceId,
+      final String kafkaClusterId
+  ) {
+    final Map<String, Object> updatedProps = new HashMap<>(props);
+    updatedProps.put(RESOURCE_LABEL_VERSION, AppInfo.getVersion());
+    updatedProps.put(RESOURCE_LABEL_COMMIT_ID, AppInfo.getCommitId());
+    updatedProps.put(RESOURCE_LABEL_TYPE, KSQL_RESOURCE_TYPE);
+    updatedProps.put(RESOURCE_LABEL_KSQL_SERVICE_ID, ksqlServiceId);
+    updatedProps.put(RESOURCE_LABEL_CLUSTER_ID, ksqlServiceId);
+    updatedProps.put(RESOURCE_LABEL_KAFKA_CLUSTER_ID, kafkaClusterId);
```

Comment:
You don't need this. According to KIP-606, "kafka.cluster.id" is for kafka broker. For client of TelemetryReporter, you only need resource.cluster.id

Reply:
removed